### DISCUSSION TITLE
Cross-linking across workspaces: Unsure if this should be a RFC or general question

### DISCUSSION THREAD

**Discussion by simkimsia** [dendronhq/dendron#1761] [General] [answered]
I have native workspaces within projects.

And also i have my own workspace that holds my daily journal notes.

I wanted to link to a particular file in a native workspace from my daily journal.

Which i can do it via file://{absolute link} but then there will be no backlinks. I don't need easy lookup or graph to show the linkages across different workspaces, but i was wondering if it would be good to include references to other workspaces vaults?

I found [workspace vault in dendron wiki](https://wiki.dendron.so/notes/6682fca0-65ed-402c-8634-94cd51463cc4/#regular-vault)

![image](https://user-images.githubusercontent.com/245021/142750799-c23e409e-1ae8-450c-bd17-bfff1d0cdd12.png)

but I am confused by how it works. so i hope somebody can enlighten me.

My current personal workspace is ~/Dendron with its own dendron.yml then i have the following project-specific dendron workspaces at

- ~/Projects/projectA/docs where dendron.yml is at ~/Projects/projectA and vault is at ~/Projects/projectA/docs
- ~/Projects/projectB/docs where dendron.yml is at ~/Projects/projectB and vault is at ~/Projects/projectB/docs

**Comment by kevinslin** [comment, marked as answer]
To answer the title, this is a general question.

To answer the actualy question:
- to link files between workspaces, you can use the `Vault Add` command to also add a workspace. this adds all the vaults within the workspace into your current workspace.  you can then create wikilinks between notes as usual
- NOTE: by default, dendron creates regular wiki links `[[link-to-note]]`. you can also create xvault links by setting [enableXVaultWikiLink](https://wiki.dendron.so/notes/3i4ABJutl7NGeXRHTnUEC.html#enablexvaultwikilink) to `true`. more on that [here](https://wiki.dendron.so/notes/3472226a-ff3c-432d-bf5d-10926f39f6c2.html#cross-vault-links)

**Reply by simkimsia** [reply]
> - to link files between workspaces, you can use the `Vault Add` command to also add a workspace. 

So I run "vault add" this at my regular dendron workspace or in the native workspace of the project?

> 
> - NOTE: by default, dendron creates regular wiki links `[[link-to-note]]`. you can also create xvault links by setting [enableXVaultWikiLink](https://wiki.dendron.so/notes/3i4ABJutl7NGeXRHTnUEC.html#enablexvaultwikilink) to `true`. more on that [here](https://wiki.dendron.so/notes/3472226a-ff3c-432d-bf5d-10926f39f6c2.html#cross-vault-links)

Same question activate this at which workspace yml?

**Reply by simkimsia** [reply]
ok i got it to work.. but I want to point out the link doesn't seem to work in preview.

## my dendron setup

- `/Users/kim/Dendron/vault` => this is my default dendron 
- `/Users/kim/Projects/projectA/docs/vault` => this is my projectA dendron vault
- `/Users/kim/Projects/projectB/docs/vault` => this is my projectB dendron vault

## What i did

inside my default dendron space

- <kbd>⌘</kbd>+<kbd>⇧</kbd>+<kbd>P</kbd> => Dendron: Vault Add
- local or remote => local
- path to another vault (relative to the current dendron vault) => `../Projects/projectA/docs/vault`
- Name this new vault => `projectA` then <kbd>⏎</kbd>
- <kbd>⌘</kbd>+<kbd>⇧</kbd>+<kbd>P</kbd> => Dendron: configure yaml
- `enableXVaultWikiLink` => from False to True

now i can `[[` and then call up notes even in the project dendron vaults.

It's too awesome for words.

Preview links don't work tho
